code/modules/vehicles/mecha/combat/honker.dm

```diff
@@ -69,13 +69,28 @@
 	fire = 100
 	acid = 100
 
-/obj/vehicle/sealed/mecha/honker/dark/add_cell(obj/item/stock_parts/cell/C)
-	if(C)
-		C.forceMove(src)
-		cell = C
+/obj/vehicle/sealed/mecha/honker/dark/add_cell(obj/item/stock_parts/cell/power_cell)
+	if(power_cell)
+		power_cell.forceMove(src)
+		cell = power_cell
 		return
 	cell = new /obj/item/stock_parts/cell/hyper(src)
 
+/obj/vehicle/sealed/mecha/honker/dark/add_scanmod(obj/item/stock_parts/scanning_module/sm)
+	if(sm)
+		sm.forceMove(src)
+		scanmod = sm
+		return
+	scanmod = new /obj/item/stock_parts/scanning_module/phasic(src)
```

Comment:
```suggestion
/obj/vehicle/sealed/mecha/honker/dark/add_scanmod(obj/item/stock_parts/scanning_module/scanmod)
	if(scanmod)
		scanmod.forceMove(src)
		src.scanmod = sm
		return
	src.scanmod = new /obj/item/stock_parts/scanning_module/phasic(src)
```
here would be a more proper way of doing this. we use `src.` to ensure it's not confused which scanmod